src/mds/Server.cc

```diff
@@ -2531,7 +2531,30 @@ void Server::dispatch_client_request(MDRequestRef& mdr)
     respond_to_request(mdr, mdr->more()->peer_error);
     return;
   }
-  
+
+  Session *session = mds->get_session(req);
+  if (session && !session->info.has_feature(CEPHFS_FEATURE_ALTERNATE_NAME) &&
+      req->get_op() != CEPH_MDS_OP_GETATTR &&
+      req->get_op() != CEPH_MDS_OP_READDIR &&
+      req->get_op() != CEPH_MDS_OP_UNLINK &&
+      req->get_op() != CEPH_MDS_OP_RMDIR) {
```

Comment:
As Jeff mentioned, we should add the `GETVXATTR` for this.  And also maybe we should allow `LSSNAP`, `RMSNAP`, `LOOKUPSNAP` and `LOOKUP*` ?

Or maybe we should allow the `RM*` and the `UNLINK` in this case, and just make it readonly in all the cases ?

Reply:
Maybe we shouldn't allow the `OPENC` too, once the old client successfully opened a file, it could get the `Fw` caps, and then it can update the file contents too ?

Reply:
> As Jeff mentioned, we should add the `GETVXATTR` for this.

Yeah, I forgot to mention this in my update: I couldn't find GETVXATTR anywhere, so I assumed this is implemented with the GETATTR.  Or am I missing something?

> And also maybe we could allow `LSSNAP`, `RMSNAP`, `LOOKUPSNAP` and `LOOKUP*` ?
> 
> Or maybe we shouldn't allow the `RM*` and the `UNLINK` in this case, and just make it readonly in all the cases ?

Right, I didn't actually tested the snapshot operations.  I'll give those a try.  I guess we should allow an admin to be able to remove *everything* (at least that was my understanding), but I'm OK either way.

Reply:
GETVXATTR is new, and may not be merged into mainline ceph yet. OPENC is already disallowed with this patchset.
We do want to allow UNLINK and RMDIR from non-fscrypted hosts. That will allow even older clients to clean out a tree (assuming they have the right creds). 

Allowing LOOKUP* makes total sense. Also allowing someone to enumerate snapshots and remove them makes sense. We don't want to allow them to create new ones though.

Reply:
Ah! I guess I'll need to update my branch then.  Ok, thanks I'll rebuild everything, try to incorporate all the suggestions in this conversation and push another revision of this patch later today after some more testing.